### PR TITLE
add [[MkString]] Operation

### PR DESCRIPTION
**Pull request by tosky001**
## What changes were proposed in this pull request?

add MkString Ops to convert a SparseTensor/DenseTensor to a Dense Tensor[String]

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by tosky001** [comment]
@yiheng  may i have your review please and thanks.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/MkString.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * MkString operation converts a SparseTensor/DenseTensor to a Dense Tensor[String]
+ *
+ * the output shape will be 1-D Tensor[String].
+ *
+ * @param strDelimiter The delimiter between key and value, default: ":"
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/MkString.scala]: looks like in the code the default is ","

**Comment by tosky001** [comment]
@yiheng done.

**Comment by yiheng** [comment]
There's a compile error
```
[ERROR] /opt/work/jenkins/workspace/BigDL-PRVN-pythontest-Spark-1.5-py27/spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/ops/MkStringSpec.scala:18: not found: object javafx
[ERROR] import javafx.scene.control.SpinnerValueFactory.DoubleSpinnerValueFactory
[ERROR]    
```

Looks like the ```DoubleSpinnerValueFactory``` is not used

**Comment by tosky001** [comment]
@yiheng  so sorry...my mistake. I import `DoubleSpinnerValueFactory` in `MkStringSpec.scala` file and the error has been modified. please run jenkins again to check it and thanks.

**Comment by yiheng** [comment]
Jenkins test pass. Merge. Thanks for the contribution.

**Pull request merged**